app/templates/components/event-card.hbs

```diff
@@ -23,8 +23,18 @@
           {{moment-format event.startTime 'ddd, MMM DD HH:mm A'}}
         </span>
       </div>
-      <div class="description">
-        {{event.shortLocationName}}
+      <div class="location">
+        {{#if device.isMobile}}
+          <div class="description" style="overflow: hidden; white-space: nowrap; text-overflow: ellipsis;">
+            {{event.shortLocationName}}
+          </div>
+        {{else if device.isTablet}}
```

Comment:
There is a lot of code duplication here, lines 28-30 are identical to lines 32-33. If the template to be rendered is same for both the conditions, why split it into two  conditions ? Please merge the two conditions